File: Src/Cxx/src/step3/Alphabet.cpp
/**
 * This file is a part of TPCI_SEDRA project
 * Copyright (c) INRIA 2015
 *
 * Contributor (s) : Maurice Bremond
 * Contacts : maurice.bremond@inria.fr
 * 
 * This project is headed by SED Rhone-Alpes service at INRIA.
 *
 */
#include "Alphabet.hpp"

#include <iostream>

bool Alphabet::IsVowel(char c)
{
  switch(c)
  {
  case 'a':
  case 'e':
  case 'i':
  case 'o':
  case 'u':
  case 'y':
  case 'A':
  case 'E':
  case 'I':
  case 'O':
  case 'U':
  case 'Y':
    return true;
  default:
    return false;
  }
}

bool Alphabet::IsConsonant(char c)
{
  return ((c >= 'A' && c <= 'Z') || (c >= 'a' && c <= 'z')) &&
    !Alphabet::IsVowel(c);
}

unsigned int Alphabet::CountVowels(const std::string& s)
{
  unsigned int counter = 0;
  for(std::string::const_iterator it = s.begin(); it != s.end(); ++it)
  {
    if (Alphabet::IsVowel(*it))
    {
      ++counter;
    }
  }
  return counter;
}

unsigned int Alphabet::CountConsonants(const std::string& s)
{
  unsigned int counter = 0;
  for(std::string::const_iterator it = s.begin(); it != s.end(); ++it)
  {
    if (Alphabet::IsConsonant(*it))
    {
      ++counter;
    }
  }
  return counter;
}

File: Src/Java/src/fr/inria/sed/icjava/Alphabet.java
/**
 * This file is a part of TPCI_SEDRA project
 * Copyright (c) INRIA 2015
 *
 * Contributor (s) : David Parsons
 * Contacts : david.parsons@inria.fr
 * 
 * This project is headed by SED Rhone-Alpes service at INRIA.
 *
 */
package fr.inria.sed.icjava;

/**
 * Created by dparsons on 13/08/15.
 */
public class Alphabet {
    public static boolean IsVowel(char c) {
        c = Character.toLowerCase(c);
        if (c == 'a')
            return true;
        if (c == 'e')
            return true;
        if (c == 'i')
            return true;
        if (c == 'o')
            return true;
        if (c == 'u')
            return true;
        if (c == 'y')
            return true;
        return false;
    }

    public static boolean IsConsonant(char c) {
        c = Character.toLowerCase(c);
        if (c < 'a')
            return false;
        if (c > 'z')
            return false;
        return (!IsVowel(c));
    }

    public static int CountVowels(String s) {
        int nbVowels = 0;
        for (char c : s.toCharArray())
            if (IsVowel(c))
                nbVowels++;
        return nbVowels;
    }

    public static int CountConsonants(String s) {
        int nbConsonants = 0;
        for (char c : s.toCharArray())
            if (IsConsonant(c))
                nbConsonants++;
        return nbConsonants;
    }
}
